openpower/package/pkg-versions.mk

```diff
@@ -0,0 +1,137 @@
+################################################################################
+#
+# pkg-versions
+#
+# Read VERSION.readme in the current directory to learn about the version
+# string structure
+#
+################################################################################
+
+define OPENPOWER_SUBPACKAGE_VERSION
+
+$(2)_VERSION_FILE = $$(OPENPOWER_VERSION_DIR)/$(1).version.txt
+ALL_SUBPACKAGE_VERSIONS += $$($(2)_VERSION_FILE)
+
+### Create subpackage patch file
+define $(2)_OPENPOWER_PATCH_FILE
+
+mkdir -p "$$(OPENPOWER_VERSION_DIR)"
+
+cd "$$(PKGDIR)"; if ls *.patch; then sha512sum *.patch | sha512sum | cut -c 1-7 | \
+	xargs echo -n > $$(OPENPOWER_VERSION_DIR)/$(1).patch.txt; \
+	else rm -f $$(OPENPOWER_VERSION_DIR)/$(1).patch.txt; fi
+
+endef
+
+### Create subpackage version file
+define $(2)_OPENPOWER_VERSION_FILE
+
+mkdir -p "$$(OPENPOWER_VERSION_DIR)"
+
+# Add package name
+echo -n "	$(1)-" > $$($(2)_VERSION_FILE)
+
+# If site local
+# Add site local and user, local commit, if local is dirty
+# Else not local
+# Add package version, op-build is dirty, op-build patches
+if [ "$$($(2)_SITE_METHOD)" == "local" ]; then \
+echo -n "site_local-" >> $$($(2)_VERSION_FILE); \
+whoami | xargs echo -n >> $$($(2)_VERSION_FILE); \
+echo -n "-" >> $$($(2)_VERSION_FILE); \
+cd "$$($(2)_SITE)"; (git describe --tags || git log -n1 --pretty=format:'%h' || echo "unknown") \
+	| sed 's/\(.*\)-g\([0-9a-f]\{7\}\).*/\2/' | xargs echo -n \
+	>> $$($(2)_VERSION_FILE); \
+cd "$$($(2)_SITE)"; git describe --all --dirty | grep -e "-dirty" | sed 's/.*\(-dirty\)/\1/' | \
+	xargs echo -n >> $$($(2)_VERSION_FILE); \
+else \
+[ `echo -n $$($(2)_VERSION) | wc -c` == "40" ] && (echo -n $$($(2)_VERSION) | \
+	sed "s/^\([0-9a-f]\{7\}\).*/\1/" >> $$($(2)_VERSION_FILE)) \
+	|| echo -n $$($(2)_VERSION) >> $$($(2)_VERSION_FILE); \
+cd "$$(BR2_EXTERNAL)"; git describe --all --dirty | \
+	if grep -e "-dirty"; then \
+	echo -n "-opbuild-dirty" >> $$($(2)_VERSION_FILE); \
+	fi; \
+if [ -f $$(OPENPOWER_VERSION_DIR)/$(1).patch.txt ]; then \
+	echo -n "-" >> $$($(2)_VERSION_FILE); \
+	cat $$(OPENPOWER_VERSION_DIR)/$(1).patch.txt >> $$($(2)_VERSION_FILE); fi \
+fi
+
+# Add new line to version.txt
+echo "" >> $$($(2)_VERSION_FILE);
+
+endef ###
+
+# Add appropriate templates to hooks
+$(2)_POST_PATCH_HOOKS += $(2)_OPENPOWER_PATCH_FILE
+$(2)_PRE_BUILD_HOOKS += $(2)_OPENPOWER_VERSION_FILE
+
+# Create rule to print subpackage version
+$(1)-version:
+	@if [ -f $$($(2)_VERSION_FILE) ]; then \
+		cat $$($(2)_VERSION_FILE); \
+		else echo "Version file does not exist, recompile needed"; fi
```

Comment:
I think it would be something like this:

if [ -n $$($(2)_VERSION_FILE) ]; then $$(($2)_OPENPOWER_VERSION_FILE); fi
cat $$($(2)_VERSION_FILE)  # file always exists now.

Since $(2)_OPENPOWER_VERSION_FILE is a set of shell commands, I don't think it would be a $(call ...)